experanto/interpolators.py

```diff
@@ -82,6 +101,17 @@ def __init__(self, root_folder: str) -> None:
                 self.start_time + np.max(self._phase_shifts),
                 self.end_time + np.min(self._phase_shifts),
             )
+        # todo - make it lazy loading? and read-only properties?
+        if "neuron_properties" in meta:
```

Comment:
I'd rather put all of this into the `Dataset` class, to keep the interpolators simple and clean. I'd rather give the dataset all the fancy properties, and keep the functionality of the interpolators to the basic.

it can look like this in the dataset:
f = self._experiment.devices["responses"].root_folder + "/meta/unit_ids.npy"
if os.exist(f):
   self.unit_ids = np.load(f)